pkg/services/sqlstore/migrations/usermig/test/user_lowercase_login_and_email_test.go

```diff
@@ -0,0 +1,253 @@
+					Created: now,
+					Updated: now,
+				},
+				{
+					ID:      2,
+					UID:     "u2",
+					Login:   "User2",
+					Email:   "USER2@domain.com",
+					Name:    "user2",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+			},
+			wantUsers: []*user.User{
+				{
+					ID:    1,
+					Login: "user1",
+					Email: "user1@domain.com",
+				},
+				{
+					ID:    2,
+					Login: "user2",
+					Email: "user2@domain.com",
+				},
+			},
+		},
+		// "2 users - same login, one already lowercase"
+		{
+			desc: "2 users with same login one already has lowercase so we keep both",
+			users: []*user.User{
+				{
+					ID:      1,
+					UID:     "u1",
+					Login:   "user1",
+					Email:   "user1@email.com",
+					Name:    "user1",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+				{
+					ID:      2,
+					UID:     "u2",
+					Login:   "User1",
+					Email:   "user1-new@email.com",
+					Name:    "user2",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+			},
+			wantUsers: []*user.User{
+				{
+					ID:    1,
+					Login: "user1",
+					Email: "user1@email.com",
+				},
+				{
+					ID:    2,
+					Login: "User1",
+					Email: "user1-new@email.com",
+				},
+			},
+		},
+		// "2 users - same login, one already lowercase"
+		{
+			desc: "2 users with same login one already has lowercase so we keep both case for uppercasing comes first in our loop",
+			users: []*user.User{
+				{
+					ID:      1,
+					UID:     "u1",
+					Login:   "User1",
+					Email:   "user1@email.com",
+					Name:    "user1",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+				{
+					ID:      2,
+					UID:     "u2",
+					Login:   "user1",
+					Email:   "user1-new@email.com",
+					Name:    "user2",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+			},
+			wantUsers: []*user.User{
+				{
+					ID:    1,
+					Login: "User1",
+					Email: "user1@email.com",
+				},
+				{
+					ID:    2,
+					Login: "user1",
+					Email: "user1-new@email.com",
+				},
+			},
+		},
+		// "2 users - same email, one already lowercase"
+		{
+			desc: "2 users with same email one already has lowercase so we keep both",
+			users: []*user.User{
+				{
+					ID:      1,
+					UID:     "u1",
+					Login:   "user1",
+					Email:   "USER1@email.com",
+					Name:    "user1",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+				{
+					ID:      2,
+					UID:     "u2",
+					Login:   "user1-new-login",
+					Email:   "user1@email.com",
+					Name:    "user2",
+					OrgID:   1,
+					Created: now,
+					Updated: now,
+				},
+			},
+			wantUsers: []*user.User{
+				{
+					ID:    1,
+					Login: "user1",
+					Email: "USER1@email.com",
+				},
+				{
+					ID:    2,
+					Login: "user1-new-login",
+					Email: "user1@email.com",
+				},
+			},
+		},
+
+		// "2 users - same login, none lowercase"
+		{
+			desc: "2 users with same login noone is lowercased we pick the most recent user to lowercase",
```

Comment:
question: how is the preference of the more recent user enforced? Is it just based on the more recently created (so the more recent entry in the DB table)?

Reply:
currently it will be the one that loops through for the user. ideally we would want to choose the one that has the latest `updated`